source/Sashimi.AzureWebApp/AzureWebAppActionHandler.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using Octopus.CoreUtilities;
+using Sashimi.AzureWebApp.Endpoints;
+using Sashimi.Server.Contracts.ActionHandlers;
+
+namespace Sashimi.AzureWebApp
+{
+    class AzureWebAppActionHandler : IActionHandlerWithAccount
+    {
+        public string Id => SpecialVariables.Action.Azure.WebAppActionTypeName;
+        public string Name => "Deploy an Azure Web App";
+        public string Description => "Deploy the contents of a package to an Azure Web App.";
+        public string? Keywords => null;
+        public bool ShowInStepTemplatePickerUI => true;
+        public bool WhenInAChildStepRunInTheContextOfTheTargetMachine => false;
+        public bool CanRunOnDeploymentTarget => false;
+        public ActionHandlerCategory[] Categories => new[] { ActionHandlerCategory.BuiltInStep, AzureConstants.AzureActionHandlerCategory, ActionHandlerCategory.Package };
+        public string[] StepBasedVariableNameForAccountIds { get; } = {SpecialVariables.Action.Azure.AccountId};
+
+        public IActionHandlerResult Execute(IActionHandlerContext context)
+        {
+            var isLegacyAction = !string.IsNullOrWhiteSpace(context.Variables.Get(SpecialVariables.Action.Azure.AccountId));
+
+            if (!isLegacyAction && context.DeploymentTargetType.Some())
+            {
+                if (context.DeploymentTargetType.Value != AzureWebAppEndpoint.AzureWebAppDeploymentTargetType)
+                    throw new InvalidOperationException($"The machine {context.DeploymentTargetName.SomeOr("<unknown>")} will not be deployed to because it is not an {AzureWebAppEndpoint.AzureWebAppDeploymentTargetType.DisplayName} target.");
+            }
+
+            return context.CalamariCommand(AzureConstants.CalamariAzure, "deploy-azure-web")
+                          .WithCheckAccountIsNotManagementCertificate(context)
+                          .WithAzureTools(context)
+                          .WithStagedPackageArgument()
+                          .Execute();
+        }
```

Comment:
The old server code had `.WithArgument("extensions", $"{AzureTools.Core},{AzureTools.AzureWebApps}")`

I think `AzureTools.Core` had the script context that logs you in https://octopusdeploy.slack.com/archives/C24PH407Q/p1596442564260500

Reply:
I'll have a look, I'll add another test

Reply:
I fixed this now @flin-8 see https://github.com/OctopusDeploy/Sashimi/pull/74/commits/50c3beedfd44b6d7983c1d67d36b9248e482a428